grpc/grpc.go

```diff
@@ -80,3 +86,69 @@ func (mi *ModuleInstance) Exports() modules.Exports {
 		Named: mi.exports,
 	}
 }
+
+// stream returns a new stream object
+func (mi *ModuleInstance) stream(c goja.ConstructorCall) *goja.Object {
+	rt := mi.vu.Runtime()
+
+	// the first argument is the GRPC client
+	obj := c.Argument(0).ToObject(rt).Export()
+
+	client, ok := obj.(*Client)
+	if !ok {
+		common.Throw(rt, errors.New("the first argument must be a GRPC client"))
+	}
+
+	// the second argument is the method name
+	methodName := sanitizeMethodName(c.Argument(1).String())
+	methodDescriptor, err := client.getMethodDescriptor(methodName)
+	if err != nil {
+		common.Throw(rt, err)
+	}
+
+	// the third argument is the params (optional)
+	// TODO: abstract logic for parsing params
+	// should be something similar to the Invoke function
+	// in js/modules/k6/grpc/client.go
+
+	registerCallback := func() func(func() error) {
+		callback := mi.vu.RegisterCallback()
+		return func(f func() error) {
+			callback(f)
+		}
+	}
+
+	tagsAndMeta := mi.vu.State().Tags.GetCurrentValues()
+
+	s := &stream{
```

Comment:
That's probably why it could not be a constructor, but I can't entirely agree that most of the fields are invited in a default way since they use the outcomes of this exact method, `stream`. I mean extracting this as the method could make part less readable with no significant benefits.